packages/playwright-core/src/server/socksClientCertificatesInterceptor.ts

```diff
@@ -150,6 +148,9 @@ class SocksProxyConnection {
         };
 
         const handleError = (error: Error) => {
+          // In case of an 'error' event on the target connection this method takes care of closing the internalTLS socket.
+          targetTLS?.removeListener('close', closeBothSockets);
```

Comment:
- I think this line is no-op, because we add a different function as an even listener.
- The comment did not really help me understand why do we need to remove the listener.